File: pi/lib/herdr_workflow/paths.py
"""Static path constants. Module-level attributes so tests/tools can override them."""
import os
from pathlib import Path

AGENT_DIR = Path.home() / ".pi" / "agent"
# pi/lib/herdr_workflow/paths.py -> herdr_workflow -> lib -> pi -> repo root
AGENT_DEF_DIR = Path(__file__).resolve().parents[3] / "agent-definitions"
CONFIG = Path(os.environ.get("HERDR_WORKFLOW_CONFIG", AGENT_DIR / "herdr-workflow.toml"))
SKILLS = AGENT_DEF_DIR / "skills"

File: agentic-coding/src/workflow/paths.ts
// Static path constants. Module-level so tests/tools can override via env.
import os from "node:os";
import path from "node:path";
import { fileURLToPath } from "node:url";

export const AGENT_DIR = path.join(os.homedir(), ".pi", "agent");

/** True when running inside a compiled bun binary (sources are bundled under a
 * virtual /$bunfs/root/, so file-relative walks break). */
export function isCompiled(): boolean {
	return (
		import.meta.url.includes("/$bunfs/") || import.meta.url.includes("B:/~BUN/")
	);
}

/** Directory where a compiled binary materializes the bundled agent-definitions. */
export const MATERIALIZED_AGENT_DEF_DIR = path.join(
	os.homedir(),
	".local",
	"share",
	"agentic-coding",
	"agent-definitions",
);

// Resolution order: explicit override -> compiled asset fallback -> source tree.
const here = fileURLToPath(import.meta.url);
export const AGENT_DEF_DIR =
	process.env.HERDR_AGENT_DEF_DIR ||
	(isCompiled()
		? MATERIALIZED_AGENT_DEF_DIR
		: path.resolve(path.dirname(here), "..", "..", "..", "agent-definitions"));

export const CONFIG =
	process.env.HERDR_WORKFLOW_CONFIG ||
	path.join(os.homedir(), ".config", "agentic-coding", "config.toml");
